plugins/node/instrumentation-react-native-navigation/src/utils/spanCreator.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AppStateStatus } from 'react-native';
+import { MutableRefObject } from 'react';
+
+import { TracerRef } from './hooks/useTrace';
+import { SpanRef } from './hooks/useSpan';
+import { Attributes, trace, context } from '@opentelemetry/api';
+
+const ATTRIBUTES = {
+  initialView: 'launch',
+  finalView: 'unmount',
+  appState: 'state.end',
+};
+
+const spanStart = (
+  tracer: TracerRef,
+  span: SpanRef,
+  currentRouteName: string,
+  customAttributes?: Attributes,
+  isLaunch?: boolean
+) => {
+  if (!tracer.current || span.current !== null) {
+    // do nothing in case for some reason the tracer is not initialized or there is already an active span
+    return;
+  }
+
+  // Starting the span
+  span.current = tracer.current.startSpan(currentRouteName);
+
+  trace.setSpan(context.active(), span.current);
+
+  // it should create the first span knowing there is not a previous view
+  span.current.setAttribute(ATTRIBUTES.initialView, !!isLaunch);
+
+  if (customAttributes) {
+    span.current.setAttributes(customAttributes);
```

Comment:
this may need to be a callback for adding vendor-specific stuff, e.g. on Embrace wanting to set `view.name` we'd need access to `currentRouteName` dynamically, alternatively it may make sense to have the instrumentation library set `view.name` instead if that seems like a useful attribute in general

Reply:
I ended up deciding to add the `view.name` as attribute as well (we can revisit in the future if we think we need something else for our integration). also just for consistency I've added the prefix `view` to other attributes like `launch`, `unmount` and `state.end` (now they are `view.launch`, `view.unmount` and `view.state.end`)